api/src/convert/archive_block.rs

```diff
@@ -138,6 +138,7 @@ mod tests {
                 target_key: RistrettoPublic::from_random(&mut rng).into(),
                 public_key: RistrettoPublic::from_random(&mut rng).into(),
                 e_fog_hint: (&[0u8; ENCRYPTED_FOG_HINT_LEN]).into(),
+                e_memo: vec![0u8; 34],
```

Comment:
Should 34 be a constant? Is a constant length going to be enforced?

Reply:
yeah it should be, that is an oversight.

i need to think some more about how to do this

Reply:
right now, there is an issue with the hashes changing because digestible crate is still changing the hash when there is a zero-length byte vector.

when you have vec of object, like, repeated message in protobuf, it makes sure that empty list doesn't add anyhting to the hash, to support schema evolution. but it seems that i didn't think to do this for vec of bytes, and that is handled as a primitive in the digestible crate code, so it would need a separate logic there.

there's two paths forward that i see for that:
(1) update digestible crate so that it doesn't hash anything for vec<u8> when the vec is empty. i'm not sure if there are other places this would have consequences. it would bring us closer to protobuf schema evolution.

(but we can't actually be completely the same as protobuf schema evolution because for things like fixed32 protobuf allows you to just add new values and old payloads will have the value defaulted, which would change the hash. so i'm not sure trying to move closer to protobuf like this is all that helpful, there are going to be discrepancies)

(2) don't use "bytes" and "vec<u8>" for e_memo field, use `Option<EncryptedMemo>` where `EncryptedMemo` is a wrapper around a fixed array, like `EncryptedFogHint`. Then use repr-bytes and derive(Message) for all that stuff, as we do for the fog hint.

the reason i didn't do (2) the first time is, there have been discussions that the custom implementations of `prost::Message` are problematic and there is talk about a change to move away from that and just have protobuf generated types and "native rust" types which are separate, and any field length enforcement would happen when converting from protobuf to native, and we would not try to fold that in somehow to the protobuf deserialization process. so if we view that stuff as tech debt now, then i was thinking to avoid writing more code like that.

(see also: https://github.com/danburkert/prost/pull/421)

OTOH that is a large effort that will happen later, and the code we have is working. so maybe (2) is the way to go, and that will also fix the digestible hash issue i think. and will allow us to enforce the length at serialization layer as we want.